presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -339,8 +346,8 @@ public Partition(
             this.recordCount = recordCount;
             this.fileCount = 1;
             this.size = size;
-            this.minValues = ImmutableMap.copyOf(requireNonNull(minValues, "minValues is null"));
-            this.maxValues = ImmutableMap.copyOf(requireNonNull(maxValues, "maxValues is null"));
+            this.minValues = new HashMap<>(requireNonNull(minValues, "minValues is null"));
```

Comment:
Wrap the copy so that the returned references cannot be modified:
```java
this.minValues = unmodifiableMap(new HashMap<>(minValues));
```

Reply:
But in this way `updateStats` will not be able to modify them neither? Did you mean to wrap them in these getter methods?

Reply:
Ah, my mistake, I didn't realize this is a mutable class. I thought we were changing from `ImmutableMap` due to having null values in the map. (which I missed in the commit message)

So ignore this comment. If this was a public class, we'd probably want to return a copy in the getter, but this isn't needed since this is a private helper class.